ir_sequence_standard_default/hooks.py

```diff
@@ -0,0 +1,29 @@
+# 2019  Vauxoo (<http://www.vauxoo.com/>)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+import logging
+from odoo import api, SUPERUSER_ID
+
+_logger = logging.getLogger(__name__)
+
+
+def post_init_hook(cr, registry):
+    env = api.Environment(cr, SUPERUSER_ID, {})
+    sequences = env['ir.sequence'].sudo().search([
+        ('implementation', '=', 'no_gap'), '|', ('active', '=', False),
+        ('active', '=', True)])
+    _logger.info("Changing sequences to Standard: %s", sequences.ids)
+    for item in sequences:
+        seq_name = "ir_sequence_%03d" % (item.id)
+        env.cr.execute("DROP SEQUENCE IF EXISTS %s" % seq_name)
+        if item.use_date_range:
+            range_ids = {}
+            for line in item.date_range_ids:
+                range_ids[line.id] = line.number_next_actual
+                seq_name = "ir_sequence_%03d_%03d" % (item.id, line.id)
+                env.cr.execute("DROP SEQUENCE IF EXISTS %s" % seq_name)
+            item.write({'implementation': 'standard'})
```

Comment:
Can remove this call to the `write` method and remove the `continue` statement

Reply:
Yes I know it could look tricky, but both lines are required, and this is why:

When you have a sequence based on Date Ranges and No Gap implementation, then you change it to Standard, all the lines at the date ranges are reset in "Next Number" to "1", in other words, the real Next Number is lost.

![Selección_783](https://user-images.githubusercontent.com/442938/69661481-e39b6580-1058-11ea-9db3-9285d88b1636.png)

![Selección_784](https://user-images.githubusercontent.com/442938/69661490-e72eec80-1058-11ea-9a29-ce64da8da27b.png)

To avoid that we first [check](https://github.com/OCA/server-tools/blob/167102097b4cf9417f4d36dc69e82cd813ebe1d7/ir_sequence_standard_default/hooks.py#L18) if it is a sequence with Date Range values, if so, we get the current value of Next Number for each line, after that, we [change](https://github.com/OCA/server-tools/blob/167102097b4cf9417f4d36dc69e82cd813ebe1d7/ir_sequence_standard_default/hooks.py#L24) the sequence to Standard (`write`), and then, we restore the right values for Next numbers on each line. On that point we call `continue` to avoid to calling the `write` outside the `if` condition, that is only executed when the Sequence is NOT based in Date Ranges.

In short words, when the sequence is Date Ranged we need to call `write` and then restore the Next Number of the sequence lines, when it is not Date Ranged, we can directly execute the `write`.

Reply:
LGTM  :+1: